applications/GeoMechanicsApplication/tests/cpp_tests/custom_conditions/test_U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -20,96 +20,353 @@
 
 namespace Kratos::Testing
 {
-
-class MockElement : public UPwSmallStrainElement<2, 4>
+class MockElement : public UPwBaseElement
 {
 public:
-    MockElement() : UPwSmallStrainElement<2, 4>() {}
+    using UPwBaseElement::CalculateOnIntegrationPoints;
 
     MockElement(IndexType NewId, const GeometryType::Pointer& rGeometry, const PropertiesType::Pointer& rProperties)
-        : UPwSmallStrainElement(NewId, rGeometry, rProperties, std::make_unique<PlaneStrainStressState>(), nullptr)
+        : UPwBaseElement(NewId, rGeometry, rProperties, std::make_unique<PlaneStrainStressState>())
     {
     }
 
     void CalculateOnIntegrationPoints(const Variable<double>& rVariable,
                                       std::vector<double>&    rOutput,
                                       const ProcessInfo&      rCurrentProcessInfo) override
     {
-        rOutput.resize(4);
+        unsigned int number_of_integration_points = 0;
+
+        // Get number of integration points based on the element type
 
+        const auto geometry_family = this->GetGeometry().GetGeometryFamily();
+
+        if (geometry_family == GeometryData::KratosGeometryFamily::Kratos_Triangle) {
```

Comment:
Sonarcloud suggests to initialize the geometry_family in the if statement to reduce the scope. 
```suggestion
        if (const auto geometry_family = this->GetGeometry().GetGeometryFamily(); 
            geometry_family == GeometryData::KratosGeometryFamily::Kratos_Triangle) {
```

However, I leave it up to you if you want to actually do this. It's always a bit debatable if this improves readability.

Reply:
It's a feature i didn't know, but also, I don't actually like it. I'll leave it as it was